src/CidrValidator.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace ipl\Validator;
+
+use ipl\I18n\Translation;
+use ipl\Stdlib\Str;
+
+/**
+ * Validate a classless inter-domain routing (CIDR)
+ */
+class CidrValidator extends BaseValidator
+{
+    use Translation;
+
+    public function isValid($value): bool
+    {
+        $this->clearMessages();
+
+        $pieces = Str::trimSplit($value, '/');
+        if (count($pieces) !== 2) {
+            $this->addMessage(sprintf($this->translate('CIDR "%s" is in the wrong format'), $value));
+
+            return false;
+        }
+
+        list($ip, $mask) = $pieces;
+        if (! filter_var($ip, FILTER_VALIDATE_IP)) {
+            $this->addMessage(sprintf($this->translate('CIDR "%s" contains an invalid IP-Address'), $value));
+
+            return false;
+        }
+
+        if (! is_numeric($mask) || $mask < 0) {
```

Comment:
Since you're working with `$prefix` (`$mask`) as int throughout the code I would cast it to int once instead of calling `is_numeric()` and reyling on type juggling.

Reply:
This test case wouldn't pass then!
https://github.com/Icinga/ipl-validator/blob/dfc8cf2a0ecc21eecd5338f96c9ccf182d2d75c7/tests/CidrValidatorTest.php#L26